src/WebJobs.Script/OutOfProc/Rpc/Configuration/WorkerConfigFactory.cs

```diff
@@ -130,13 +130,12 @@ internal void AddProvider(string workerDir)
                 workerDescription.Arguments = workerDescription.Arguments ?? new List<string>();
                 GetDefaultExecutablePathFromAppSettings(workerDescription, languageSection);
                 AddArgumentsFromAppSettings(workerDescription, languageSection);
-                workerDescription.ApplyDefaultsAndValidate();
-                _workerDescripionDictionary[workerDescription.Language] = workerDescription;
-
                 if (IsHydrationNeeded(workerDescription.DefaultWorkerPath))
                 {
                     workerDescription.DefaultWorkerPath = GetHydratedWorkerPath(workerDescription);
                 }
+                workerDescription.ApplyDefaultsAndValidate();
```

Comment:
This is a bit out of scope for this change. Is this required in order to get your tests to work? If so, we might want to consider moving those tests to a different PR or at least keeping them in a different commit

Reply:
Yes, it's needed for the tests. We need to populated the default worker path with the proper values for the worker file to be found.